modules/porous_flow/src/materials/PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity.C

```diff
@@ -0,0 +1,75 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity.h"
+
+template <>
+InputParameters
+validParams<PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity>()
+{
+  InputParameters params = validParams<PorousFlowTotalGravitationalDensityBase>();
+  params.addRequiredRangeCheckedParam<Real>(
+      "rho_s", "rho_s > 0", "The density of the solid matrix");
+  params.addClassDescription(
+      "This Material calculates the porous medium density from the porosity, solid density "
+      "(assumed "
+      "constant) and fluid density, for the fully-saturated case, using a linear weighted average. "
+      "density = phi * rho_f + (1 - phi) * rho_s, where phi is porosity and rho_f, rho_s are the "
+      "densities of "
+      "the fluid and solid phases.");
+  return params;
+}
+
+PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity::
+    PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity(
+        const InputParameters & parameters)
+  : PorousFlowTotalGravitationalDensityBase(parameters),
+    _rho_s(getParam<Real>("rho_s")),
+    _rho_f_qp(getMaterialProperty<std::vector<Real>>("PorousFlow_fluid_phase_density_qp")),
+    _porosity_qp(getMaterialProperty<Real>("PorousFlow_porosity_qp")),
+    _drho_f_qp_dvar(getMaterialProperty<std::vector<std::vector<Real>>>(
+        "dPorousFlow_fluid_phase_density_qp_dvar")),
+    _dporosity_qp_dvar(getMaterialProperty<std::vector<Real>>("dPorousFlow_porosity_qp_dvar"))
+{
+  if (_nodal_material == true)
+    mooseError("PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity is only defined for "
+               "at_nodes = false");
+}
+
+Real
+PorousFlowTotalGravitationalDensityFullySaturatedFromPorosity::rho_f(unsigned int ph) const
+{
+  return _rho_f_qp[_qp][ph];
```

Comment:
If you follow my suggestion above, this line will just be
```
return _rho_f_qp[_qp][0];
```
in your case, and it'll be ready to be overridden by someone else.